test/vcf/debugulator_integration_test.cpp

```diff
@@ -52,79 +52,124 @@ namespace ebi
+      // fixed_vcf: writing result vcf to a stringstream in memory
+      file.clear();
+      file.seekg(0);
+      report->flush();
+      std::stringstream fixed_vcf;
+      vcf::debugulator::fix_vcf_file(file, *report, fixed_vcf);
+      file.close();
+      long fixed_vcf_lines = count_lines(fixed_vcf);
+
+
+      // here we check that the fixed file is valid:
+      // fixed_vcf: stringstream the debugulator::fix_vcf_file wrote
+      // reports: ReportWriters (one OdbReportRW writing to database "errors_after")
+      fixed_vcf.seekg(0);
+      auto validator2 = ebi::vcf::build_parser(path.string(), ebi::vcf::ValidationLevel::warning, version);
+      auto db_path_after = boost::filesystem::path{"/tmp/"} / path.filename();
+      db_path_after += ".debugulator_test.errors_after.db";
+      boost::filesystem::remove(db_path_after);   // make sure the dbs doesn't exist from previous runs
+      auto output_after = new ebi::vcf::OdbReportRW{db_path_after.string()};
+      reports.clear();
+      reports.emplace_back(output_after);
+      bool successfully_fixed = is_valid_vcf_file(fixed_vcf, *validator2, reports);
+
+      auto fixed_vcf_path = boost::filesystem::path{"/tmp/"} / path.filename();
+      fixed_vcf_path += ".debugulator_test.fixed.vcf";
+      boost::filesystem::remove(fixed_vcf_path);
+      std::stringstream debug_message;
+      if (not successfully_fixed) {
+          // dump fixed file to manually inspect
+          std::ofstream fixed_file{fixed_vcf_path.string()};
+          fixed_file << fixed_vcf.str();
+          fixed_file.close();
+          debug_message << "This test failed, you can see the related data at: \n" << db_path_after.string() << "\n";
+          debug_message << db_path.string() << "\n" << fixed_vcf_path.string() << std::endl;
+      } else {
+          // clean report dbs
+          boost::filesystem::remove(db_path);
+          boost::filesystem::remove(db_path_after);
+      }
+      return std::tuple<bool, bool, std::string, long>{first_validation,
+                                                       successfully_fixed,
+                                                       debug_message.str(),
+                                                       fixed_vcf_lines};
+  }
+
+  TEST_CASE("Fixing a VCF with duplicates", "[debugulator]")
+  {
+      ebi::vcf::Version version = ebi::vcf::Version::v41;
+      bool first_validation, validation_after_fixing;
+      std::string debug_message;
+      long lines;
+      for (size_t i = 1; i <= 3; ++i) {
+          auto path = boost::filesystem::path("test/input_files/v4." + std::to_string(i) + "/failed/failed_body_duplicated_001.vcf");
+          SECTION(path.string()) {
+              std::tie(first_validation, validation_after_fixing, debug_message, lines) = fix_file(path, version);
+              CHECK_FALSE(first_validation);
+              INFO(debug_message);
+              CHECK(validation_after_fixing);
+              CHECK(lines == 6);
```

Comment:
These checks should be embedded into the testing function, otherwise it could run for a long time under incorrect assumptions (for instance, if `first_validation` is true is should fail straight away).

Splitting them in multiple functions would also improve readability.